x-pack/platform/plugins/shared/fleet/common/types/models/fleet_server_policy_config.ts

```diff
@@ -4,15 +4,29 @@
  * 2.0; you may not use this file except in compliance with the Elastic License
  * 2.0.
  */
+import type { SOSecret } from './secret';
 
-// SO definition for this type is declared in server/types/interfaces
 export interface NewFleetServerHost {
   name: string;
   host_urls: string[];
   is_default: boolean;
   is_preconfigured: boolean;
   is_internal?: boolean;
   proxy_id?: string | null;
+  ssl?: {
+    certificate_authorities?: string[];
+    certificate?: string;
+    key?: string;
+    es_certificate_authorities?: string[];
+    es_certificate?: string;
+    es_key?: string;
+  } | null;
+  secrets?: {
```

Comment:
Are these secrets being used by Fleet-Server/Agent from the agent policy or only in the install command? 
If only in the command, using secrets might not be needed.

Also, secrets can't be read by kibana, only Fleet-server, so it's going to be problematic to show them in the UI instructions.

Reply:
> Are these secrets being used by Fleet-Server/Agent from the agent policy or only in the install command?

I think that it is required to pass the secrets directly to fleet server, but I'll do those changes in follow up PR as this one it's already big.

> Also, secrets can't be read by kibana, only Fleet-server, so it's going to be problematic to show them in the UI instructions. 

The actual secrets are the cert keys and those are not shown in the instructions. I based this implementation on the outputs one, do you think I should change the data structure in a different way?

Reply:
Okay, if not needed to show in the instructions, it should be good. 
I guess it's good like this, let's just make sure the secrets are used correctly in fleet-server/agent and test end to end when the feature is ready.

Reply:
@criamico an @juliaElastic I'm not concerned with the certs added at installation (day 0), this feature is mainly for life-cycle management of these entities (day 1 onwards). The fact of the matter is that when the fleet server is installed, the cli options we have today will have to be used to get the right certificates to be loaded. The UI options on the onboarding pages won;t add much value in that case.  In fact an advanced user won't be using the onboarding.  I'm happy for this to be revisited if indeed we see cases where it's an issue for our users. 

I sense that is the decision you came up with any way :-)